tests/integration/test_interface.py

```diff
@@ -3,80 +3,181 @@
 
     # Configure if default (sample data) is used or a different root directory
     if use_sample_data:
         source = None  # will default to using sample data
     else:
-        source = _generate_non_default_data_root(
-            data_dir_name, tmpdir, sample_data_path, data
-        )
-
-    noised_data = noising_function(seed=0, source=source)
-    noised_data_same_seed = noising_function(seed=0, source=source)
-    noised_data_different_seed = noising_function(seed=1, source=source)
+        source = _generate_non_default_data_root(dataset.name, tmpdir, sample_data_path, data)
+
+    # Update unit number cell probabilities because there are so few of them
+    # that nothing gets noised with default levels
+    config = {
+        dataset.name: {
+            Keys.COLUMN_NOISE: {
+                column.name: {
+                    noise_type.name: {
+                        Keys.CELL_PROBABILITY: 0.25,
+                    }
+                    for noise_type in COLUMNS.unit_number.noise_types
+                }
+                for column in dataset.columns
+                if "unit_number" in column.name
+            },
+        },
+    }
+
+    # Update SSA dataset to noise 'ssn' but NOT noise 'ssa_event_type' since that
+    # will be used as an identifier along with simulant_id
+    # TODO: Noise ssa_event_type when record IDs are implemented (MIC-4039)
+    if dataset.name == DATASETS.ssa.name:
+        config[dataset.name][Keys.COLUMN_NOISE][COLUMNS.ssn.name] = {
+            noise_type.name: {
+                Keys.CELL_PROBABILITY: 0.01,
+            }
+            for noise_type in COLUMNS.ssn.noise_types
+        }
+        config[dataset.name][Keys.COLUMN_NOISE][COLUMNS.ssa_event_type.name] = {
+            noise_type.name: {
+                Keys.CELL_PROBABILITY: 0,
+            }
+            for noise_type in COLUMNS.ssa_event_type.noise_types
+        }
+
+    # FIXME: Do not set the row-level probability to zero
+    # Set row-level noise to zero since we're just testing columns here.
+    config[dataset.name][Keys.ROW_NOISE] = {
+        noise_type.name: {
+            Keys.ROW_PROBABILITY: 0,
+        }
+        for noise_type in dataset.row_noise_types
+    }
+
+    noised_data = noising_function(seed=0, source=source, year=None, config=config)
+    noised_data_same_seed = noising_function(seed=0, source=source, year=None, config=config)
+    noised_data_different_seed = noising_function(
+        seed=1, source=source, year=None, config=config
+    )
 
     assert not data.equals(noised_data)
     assert noised_data.equals(noised_data_same_seed)
     assert not noised_data.equals(noised_data_different_seed)
 
-    # Check each column's dtype
-    for col in noised_data.columns:
-        expected_dtype = [c.dtype_name for c in COLUMNS if c.name == col][0]
+    # Check each column. We set the index for each dataset to be unique
+    # identifiers b/c the original index gets reset after noising. Note that
```

Comment:
Where does the index get reset?  At the dataset level?  Not in the noise function for columns.

Reply:
https://github.com/ihmeuw/pseudopeople/blob/main/src/pseudopeople/interface.py#L78